Sources/MongoKitten/Aggregate.swift

```diff
@@ -104,6 +104,12 @@ public struct AggregateBuilderPipeline: QueryCursor {
         let command = makeCommand()
         let connection = try await getConnection()
         
+        let minimalVersionRequired = stages.compactMap { $0.minimalVersionRequired }.min()
+        
+        if let actualVersion = connection.wireVersion, let minimalVersion = minimalVersionRequired, actualVersion < minimalVersion {
+            connection.logger.warning("Aggregation might fail since one or more aggregation stages require a higher MongoDB version than provided by the current connection.")
```

Comment:
Perfect! Just one tiny change. I don't think you'll want this error to repeat for every query, or at least you'll want to be able to silence this. So I'd wrap this statement in `#if DEBUG` so that these logs are only visible in development environments.